src/net.cpp

```diff
@@ -51,8 +52,12 @@
 #include <array>
 #include <cstdint>
 #include <functional>
+#include <list>
```

Comment:
In 6e56589bcbe2d04ee1b9018baf5a9d1dc958e68d

> The compiler raises when they are removed from src/sync.h in the last commit of this PR, so they need to be added.

If they need to be added, they should be added in the commit where they become required. If this PR is going to be backported, the number of commits, and potential for conflicts should be kept at a minimum, so we should avoid things like additional commits doing include fixups.

Reply:
These are different changes; kept them separate to ease review. If the goal is to squash this down, then we could not conditionally do the two includes in the 3rd commit and backport only that one.

Reply:
Minimal version in #24770 for backport.